regular_expressions_intermediate/regular_expressions_intermediate.md

```diff
@@ -0,0 +1,540 @@
+
+![The regular expression shows three matches, each of which has two captured groups.](media/moveahead.png) <!-- style = "border: 1px solid rgb(var(--color-highlight)); " -->
+
+This is a case where you would like the regex engine to not **move** ahead, but rather **look** ahead.  Imagine that the regex engine moves along and finds 20.3.  Then it keeps its place and doesn't move, but it "looks" ahead and sees that yes, 20.3 is indeed followed by another decimal number, so this is a match.  The computer therefore captures that pair, 20.3 and 22.0.  Once that's done, it starts moving again, to 22.0, and when it hits the space after 22.0, it again stops, and **looks** ahead, without **moving** ahead, to see 19.5.  That's our second pair of numbers, 22.0 and 19.5.  This continues and we manage to find all 5 pairs of consecutive numbers, even when the patterns overlap.  
+
+The **lookahead** regular expression syntax takes the form of `(?=regex)` where `regex` is replaced with what you're looking for.  So, for example, our pattern for the second decimal number, which is currently in a capturing group and looks like: `([\d\.]+)` could be placed between `(?=` and `)`, giving us `(?=([\d\.]+))`.
+Go ahead and [try that in a regex checker](https://regex101.com/r/NvzPb0/1) and you should see something like the following:
+
+![The regular expression shows five matches, each of which has two captured groups.](media/lookahead.png) <!-- style = "border: 1px solid rgb(var(--color-highlight));" -->
+
+<div class = "behind-the-scenes">
+<b style="color: rgb(var(--color-highlight));">Behind the scenes</b><br>
+
+There are other types of "lookarounds", including **lookbehinds**, that you might consider learning if you write a lot of regular expressions, but for now, the lookahead by itself will serve you well for more complex cases.
+
+</div>
+
+## Quiz 4: Anchors, Boundaries, and Lookaheads
+
+What's the anchor character that indicates the "start of a string"?
+
+[[^]]
+<script>
+  let input = "@input".trim();
+  input == "^";
+</script>
+***
+<div class = "answer">
+
+The anchor for the start of a string is the caret, or `^`, symbol.
+
+</div>
+***
+
+What's the anchor character that indicates the "end of a string"?
+
+[[$]]
+<script>
+  let input = "@input".trim();
+  input == "$";
+</script>
+***
+<div class = "answer">
+
+The anchor for the start of a string is the dollar sign, or `$`, symbol.
+
+</div>
+***
+
+What's the two character sequence that indicates the boundary between a word-type character (like a letter, number, or underscore) and a non-word-type character (like punctuation or spaces)?
+
+[[\b]]
+<script>
+  let input = "@input".trim();
+  input == "\b";
+</script>
+***
+<div class = "answer">
+
+The code for a word boundary is ` \b `.
```

Comment:
```suggestion
The code for a word boundary is `\b`.
```

Reply:
agree, added in new commit, not sure why I did it this way????